---
title: "RNAseq iPSC microglia"
author: "Alicia Pliego"
date: "3/15/2022"
output: html_document
runtime: shiny
---

# RNAseq BRAF analysis

## Research questions

	1- Are these iPSC expressing human-like microglia genes?


```{r setup, include=FALSE, echo=FALSE, message=FALSE}
library("DESeq2")
library(tximport)
library(plotly)
library(dplyr) 
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# txi import quant files by genes from salmon into DESEQ2

counts <- read.table('/Users/aliciapliego/Projects/iPSC/data/Count_QC-raw-count_v2.txt', header = TRUE)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
counts_nondup <- subset(counts, !duplicated(subset(counts, select=c(gene_name))))
rownames(counts_nondup) <- counts_nondup$gene_name
counts_nondup$gene_name <- NULL
```


```{r,  echo=FALSE, warning=FALSE, message=FALSE}
metadata =readxl::read_excel('/Users/aliciapliego/Projects/iPSC/data/metadata.xlsx')

fig <- plot_ly(
  type = 'table',
  header = list(
    values = c("<b>Samples</b>", names(metadata)),
  align = c('left', rep('center', ncol(metadata))),
  line = list(width = 1, color = 'black'),
  fill = list(color = 'rgb(235, 100, 230)'),
  font = list(family = "Arial", size = 14, color = "white")
  ),
  cells = list(
    values = rbind(
      rownames(metadata), 
      t(as.matrix(unname(metadata)))
    ),
    align = c('left', rep('center', ncol(metadata))),
    line = list(color = "black", width = 1),
    fill = list(color = c('rgb(235, 193, 238)', 'rgba(228, 222, 249, 0.65)')),
    font = list(family = "Arial", size = 12, color = c("black"))
  ))

fig
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

dds <- DESeqDataSetFromMatrix(countData = round(counts_nondup),
                              colData = metadata,
                              design = ~ Type)
```

### Visualizing library sizes 
Sample reads are usually lower after normalization and gene filtering done in previous steps after count matrix generation.
```{r, echo=FALSE, warning=FALSE, message=FALSE}
counts <- colSums(assay(dds))
counts <- as.data.frame(counts)
rownames(counts) <- metadata$Name
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}

hline <- function(y = 20000000, color = "red") {
  list(
    type = "line",
    x0 = 0,
    x1 = 1,
    xref = "paper",
    y0 = y,
    y1 = y,
    line = list(color = color)
  )
}
fig <- plot_ly( counts, x= rownames(counts), y= counts$counts,
  name = "Barplot of library sizes",
  type = "bar" )%>%
  layout(shapes = list(hline(20000000), list(type = "rect",line = list(color = "black"),
                                        x0 = 0.9, x1 = 2)), plot_bgcolor = "#e5ecf6", title = "Barplot of library sizes",
         xaxis = list(title = 'Samples'), 
         yaxis = list(title = 'Million of reads'))

fig
```

### Filtering non-expressed genes
- Remove genes with a count lower than 5
Plot the number of samples a gene is expressed in:
```{r,echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
is_expressed <- assay(dds) >= 5
hist(rowSums(is_expressed),main="Number of samples a gene is expressed in",xlab="Sample Count")
```

It seems that genes are either expressed in all samples, or not expressed at all. Only keep genes that are expressed in at least 2 samples.

- Filter genes that are expressed in at least 2 samples.

### Visualize count distributions
Applying the boxplot function to the raw counts from our dataset reveals something about the nature of the data; the distributions are dominated by a few genes with very large counts.
```{r,echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
keep <- rowSums(assay(dds) >= 5) >= 2
dds <- dds[keep,]
boxplot(assay(dds))

```


```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
boxplot(log10(assay(dds)))

```

Log2 scale data with the vsd function from DEseq2 to remove the dependence of the variance on the mean, particurarly the high variance of the logarithm of count data when the mean is low. 
```{r,include=FALSE, echo=FALSE, warning=FALSE, message=FALSE}
# Get log2 counts
vsd <- vst(dds,blind=TRUE)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
# Check distributions of samples using boxplots
boxplot(assay(vsd), xlab="", ylab="Log2 counts per million",las=2,main="Normalised Distributions")
# Let's add a blue horizontal line that corresponds to the median logCPM
abline(h=median(assay(vsd)), col="blue")
```

### Heatmap of the sample-to-sample distances 
I applied the dist function to the transpose of the transformed count matrix to get sample-to-sample distances.

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
sampleDists <- dist(t(assay(vsd)))
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
library(RColorBrewer)
library(pheatmap)
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(colData(dds)$Name, colData(dds)$Status, sep="-")
colnames(sampleDistMatrix) <- colData(dds)$Name
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         col=colors)
```

As we already expected, all samples are highly correlated expect tw047.

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="Type")

```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="Name")
```

We showed with the correlation heatmap and the PCA that sample tw047 is not correlated to the other groups. Also, samples from tw029 nicely cluster together and don't differ from av111 which shows a similar trend. However, one of the samples is a little bit different (av111_001) which might be due to experimental changes..
